design/ErrorHandling.md

```diff
@@ -169,7 +174,14 @@ It is only necessary if an application wishes to recover from recoverable errors
    - <mesh3.indexBuffer> is invalid because it got an unsupported usage flag (0x89)
    ```
 
- - The exact shape of `WebGPUObjectStatusQuery` (currently `Promise<WebGPUObjectStatus>`) may piggy-back on the decision taken for `WebGPUFence`.
+ - Should applications be able to intentionally create graphs of potentially-invalid objects, and recover from this late?
+   E.g. create a large buffer, create a bind group from that, create a command buffer from that, then choose whether to submit based on whether the buffer was successfully allocated.
+    - If yes, `tryCreateBuffer` must return `GPUBuffer` and error log entries must not be generated when creating objects from invalid objects.
+      (Only log errors on queue.submit and other device/queue level operations.)
+    - If no, `tryCreateBuffer` should return `Promise<GPUBuffer>` and error log entries should be generated when creating objects from invalid objects.
```

Comment:
+1 for this because it seems simpler both conceptually and for the implematation.

Reply:
I still kind of like the idea of errors happening only on submit et al, but I agree - I don't think this use-case is very important.

(I don't see why the implementation would be simpler, though - we still have to allow graphs of potentially invalid objects.)